xls/modules/rle/rle_enc_adv.md

```diff
@@ -0,0 +1,85 @@
+# Multisymbol Run-length Encoder
+
+The encoder uses Run Length Encoding (RLE) to compress the input stream of
+repeating symbols to the output stream that contains the symbols and
+the number of its consecutive occurrences in the input stream.
+
+Overall, we can break down the data processing into four stages: reduction, alignment, compression, and output generation. The division of responsibility allowed the specialized blocks to efficiently process data and made it possible test each functionality separately.
```

Comment:
did we measure the overhead of splitting the process into different proc in term of additional registers and wires, compared to a single proc implementation?

Reply:
I haven't checked how much more resource does multi process implementation uses over single process implementation, and how max frequency is affected by this.